pkg/cmd/roachtest/test_runner.go

```diff
@@ -137,6 +137,9 @@ type testRunner struct {
 		skip    map[*testImpl]struct{}
 	}
 
+	// testCount is the total number of tests registered multiplied by the total number of times each test has to run.
+	testCount int
```

Comment:
Nit: I don't think we need a new struct field for this? Maybe `runWorker` could take a `maxTotalFailures` parameter, since the value is only used in that context.

Also, `testCount` is also deceptively similar sounding to our `roachtest --count` parameter, which is different (albeit related) number.

Reply:
Hmm, makes sense. Will update.

Reply:
Done